lib/Parser/Parse.cpp

```diff
@@ -10106,7 +10115,7 @@ ParseNodePtr Parser::ParseStatement()
     {
         if (buildAST)
         {
-            if (nullptr == m_currentNodeFunc)
+            if (nullptr == m_currentNodeFunc || IsImportOrExportStatementValidHere())
```

Comment:
This looks weird to me. 
Is this saying that `return` is an error, but 'try { return; } catch {}` is not?